asllib/doc/Types.tex

```diff
@@ -486,8 +610,52 @@ \subsubsection{Example}
 \end{mathpar}
 \CodeSubsection{\TBoolBegin}{\TBoolEnd}{../Typing.ml}
 
-\hypertarget{bitvectortypeterm}{}
 \section{Bitvector Types\label{sec:BitvectorTypes}}
+\hypertarget{bitvectortypeterm}{}
+\emph{Bitvectors} represent sequences of $0$ and $1$.
+%
+The \texttt{bits(N)} type represents a bitvector of length \texttt{N},
+where \texttt{N} may specify a fixed width or a constrained width.
+%
+The \texttt{bit} type is syntactic sugar for \texttt{bits(1)} (see \ASTRuleRef{Ty.TBits}.\texttt{BIT}).
+
+\identi{KGMC}%
+The syntax for \bitvectortypesterm{} has an optional $\Nbitfields$,
+which allows specifying \emph{\bitfieldsterm} ---
+\bitslicesterm{} of bitvectors --- to be treated as named
+fields that can be read or written.
+\chapref{Bitfields} defines \bitfieldsterm{}
+and \chapref{BitvectorSlicing} defines \bitslicesterm{}.
+
+\RequirementDef{BitvectorWidthImmutable}
+The width of a bitvector cannot be modified.
+
+\RequirementDef{BitvectorWidthBounds}
+There is no bound on the maximum bitvector width allowed, although an implementation may specify an upper
+limit.
+%The minimum bound is zero.
+It is recognized that zero-length bitvectors might not be supported in systems
+to which ASL might be translated (such as SMT solvers),
+and an implementation might need to lower bitvector
+expressions to a form where they do not exist.
```

Comment:
please disambiguate "they" in the phrase "where they do not exist": I suppose this is "zero-length bit vectors"